ios/app/watchos/extension/InterfaceController.swift

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright @ 2018-present 8x8, Inc.
+ * Copyright @ 2017-2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import WatchKit
+import Foundation
+
+
+class InterfaceController: WKInterfaceController {
+
+    @IBOutlet var table: WKInterfaceTable!
+
+    func updateUI(_ newContext:JitsiMeetContext) {
+        if let recentURLsArray = newContext.recentURLs {
+            updateRecents(withRecents:  recentURLsArray, currentContext: newContext)
+        }
+    }
+
+    private func updateRecents(withRecents recents: NSArray, currentContext: JitsiMeetContext) {
+        // Updating the # of rows only if it actually changed prevents from blinking the UI
+        if (table.numberOfRows != recents.count) {
+            table.setNumberOfRows(recents.count, withRowType: "MeetingRowType")
+        }
+
+        for (index, entry) in recents.enumerated() {
+            let entryDict = entry as! NSDictionary
+            let roomURL = entryDict["conference"] as! NSString
+            let timestamp = entryDict["date"] as! NSNumber
+
+            // Prepare values
+            let room = roomURL.components(separatedBy: "/").last
```

Comment:
come to think of it, does it make sense to send the full URL to the watch instead of just the room? looks like it would simplify things a bit.

Reply:
That requires pre-processing the recent list on the JavaScript side, so not sure if that would be simpler. Also the URL is used in the `joinConference` command sent by the watch and the full URL is needed there to work correctly

Reply:
👍  Yeah, I noticed it later and forgot to delete this comment.